scripts/publish_grok46_hub.py

```
@@ -0,0 +1,811 @@
+    """Return notes that correspond to exactly one visible published batch."""
+    labels = {label[2] for batch in batches if (label := batch_label(batch)) is not None}
+    if any(batch.name in LEGACY_R1_NAMES for batch in batches):
+        labels.add("r01")
+    mode_path = src / ".round-marker-mode.json"
+    baseline = None
+    manifests = {}
+    if mode_path.is_file():
+        try:
+            mode = json.loads(mode_path.read_text())
+        except (OSError, json.JSONDecodeError) as exc:
+            raise SystemExit(f"cannot read marker mode for {src}: {exc}") from exc
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)